lib/ex_doc/markdown.ex

```diff
@@ -33,21 +33,32 @@ defmodule ExDoc.Markdown do
   Converts the given markdown document to HTML AST.
   """
   def to_ast(text, opts \\ []) when is_binary(text) do
-    get_markdown_processor().to_ast(text, opts)
+    {processor, options} = get_markdown_processor()
+    processor.to_ast(text, options |> Keyword.merge(opts))
   end
 
   @doc """
   Gets the current markdown processor set globally.
   """
   def get_markdown_processor do
-    case Application.fetch_env(:ex_doc, @markdown_processor_key) do
-      {:ok, processor} ->
-        processor
+    {processor, options} = case Application.fetch_env(:ex_doc, @markdown_processor_key) do
```

Comment:
Sorry, one last nitpick: let's normalize it when we write (on put_markdown_process) and not when we read. It is generally preferred to normalize data when we store it, instead of every time we write. :)